native/dlantr.go

```diff
@@ -0,0 +1,248 @@
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/lapack"
+)
+
+// Dlantr computes the specified norm of an m×n trapezoidal matrix A. If
+// norm == lapack.MaxColumnSum work must have length at least n, otherwise work
+// is unused.
+func (impl Implementation) Dlantr(norm lapack.MatrixNorm, uplo blas.Uplo, diag blas.Diag, m, n int, a []float64, lda int, work []float64) float64 {
+	checkMatrix(m, n, a, lda)
+	switch norm {
+	case lapack.MaxRowSum, lapack.MaxColumnSum, lapack.NormFrob, lapack.MaxAbs:
+	default:
+		panic(badNorm)
+	}
+	if uplo != blas.Upper && uplo != blas.Lower {
+		panic(badUplo)
+	}
+	if diag != blas.Unit && diag != blas.NonUnit {
+		panic(badDiag)
+	}
+	if norm == lapack.MaxColumnSum && len(work) < n {
+		panic(badWork)
+	}
+	if min(m, n) == 0 {
+		return 0
+	}
+	switch norm {
+	default:
+		panic("unreachable")
```

Comment:
Assuming the comment above is not something that can change, this is here to assert that nothing bad has been done to the code? Otherwise just delete the default?